stablehlo/dialect/StablehloOps.td

```diff
@@ -2660,24 +2660,27 @@ def StableHLO_ReshapeOp: StableHLO_Op<"reshape",
 }
 
 def StableHLO_DynamicReshapeOp: StableHLO_ShapedInterfaceOp<"dynamic_reshape",
-      [HLO_SpeculatableIfAllInputsStaticAndShapeConstant, NoMemoryEffect]> {
+      [HLO_SpeculatableIfAllInputsStaticAndShapeConstant, NoMemoryEffect,
+       AllElementTypesMatch<["operand", "result"]> /*dynamic_reshape_c1*/]> {
```

Comment:
finding a best way to relax this check for per-axis quantized element type. 

`HLO_CompatibleOperandsAndResultElementType or SameOperandsAndResultElementType ` Traits try to match all  operands including `output_shape`, which is expected to mismatch.

Reply:
since existing Traits are not useful,  moving `element_type(result)  =  element_type(operand` verification logic  to the op verifier.